src/framework/audio/internal/mixer.cpp

```diff
@@ -0,0 +1,241 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "mixer.h"
+#include "log.h"
+
+using namespace mu::audio;
+
+Mixer::Mixer()
+{
+    buildRpcReflection();
+}
+
+Mixer::~Mixer()
+{
+}
+
+Mixer::Mode Mixer::mode() const
+{
+    return m_mode;
+}
+
+void Mixer::setMode(const Mixer::Mode& mode)
+{
+    m_mode = mode;
+}
+
+void Mixer::setClock(std::shared_ptr<Clock> clock)
+{
+    m_clock = clock;
```

Comment:
thread-unsafe code
this is the method called from the main thread, but `m_clock` used in the worker thread